test/unit/org/apache/cassandra/db/partitions/SimplePartition.java

```diff
@@ -0,0 +1,292 @@
+        {
+            if (!column.type.unwrap().isMultiCell())
+                throw new IllegalArgumentException("Unable to add multiple values to a regular column " + column);
+            builder.addComplexDeletion(column, DeletionTime.build(timestamp - 1, timestamp - 1));
+            // map needs to be specially handled as its key/value
+            if (column.type.unwrap() instanceof MapType)
+            {
+                for (int i = 0; i < values.size(); i = i + 2)
+                {
+                    ByteBuffer key = values.get(i);
+                    ByteBuffer value = values.get(i + 1);
+                    builder.addCell(BufferCell.live(column, timestamp, value, CellPath.create(key)));
+                }
+            }
+            else
+            {
+                for (int i = 0; i < values.size(); i++)
+                    builder.addCell(complexCell(column, i, values.get(i), timestamp));
+            }
+            return this;
+        }
+
+        private Cell<?> complexCell(ColumnMetadata column, int idx, ByteBuffer value, long timestamp)
+        {
+            var type = column.type.unwrap();
+            if (type.isCollection())
+            {
+                CollectionType<?> ct = (CollectionType<?>) type;
+                switch (ct.kind)
+                {
+                    case SET:
+                    {
+                        // this isn't correct... the value isn't actually known... so only support map with key/value matching...
+                        return BufferCell.live(column, timestamp, value, CellPath.create(value));
+                    }
+                    case LIST:
+                    {
+                        // this isn't actually correct, as the cellpath is based off time, but a counter is used to keep things deterministic
+                        CellPath path = CellPath.create(ByteBuffer.wrap(TimeUUID.Generator.atUnixMillisAsBytes(idx)));
+                        return BufferCell.live(column, timestamp, value, path);
+                    }
+                    case MAP:
+                        throw new UnsupportedOperationException("Map isn't supported due to API being single element rather than multi-element");
+                    default:
+                        throw new UnsupportedOperationException(ct.kind.name());
+                }
+            }
+            else if (type.isUDT())
+            {
+                UserType ut = (UserType) type;
+                CellPath path = ut.cellPathForField(ut.fieldName(idx));
+                return BufferCell.live(column, timestamp, value, path);
+            }
+
+            throw new UnsupportedOperationException(type.toString());
+        }
+
+        public SimplePartition build()
+        {
+            SimplePartition.this.add(builder.build());
+            return SimplePartition.this;
+        }
+    }
+
+    private class Updater implements UpdateFunction<Row, Row>, ColumnData.PostReconciliationFunction
```

Comment:
honestly copied a ton from atomic btree partition, the issue i faced with reusing is having an allocator...  I could try to make that logic reusable but didn't want to touch a critical section of code so I forked

Reply:
switched to `AtomicBTreePartition` and the test is passing at least